staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go

```diff
@@ -100,3 +94,53 @@ func WithLateConnectionFilter(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, r)
 	})
 }
+
+// WithRejectBeforeReady rejects the request until the process has been ready once.
+func WithRejectBeforeReady(handler http.Handler, hasBeenReadyCh <-chan struct{}, s runtime.NegotiatedSerializer) http.Handler {
+	var nonReadyRequestReceived atomic.Bool
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if hasBeenReadyCh != nil {
+			if _, closed := <-hasBeenReadyCh; closed {
+				handler.ServeHTTP(w, r)
+				return
+			}
+		}
+
+		// ignore connections to local IP. Those clients better know what they are doing.
+		if pth := "/" + strings.TrimLeft(r.URL.Path, "/"); pth != "/readyz" && pth != "/healthz" && pth != "/livez" {
+			if isLocal(r) {
+				klog.V(2).Infof("Loopback request to %q (user agent %q). This client probably does not watch /readyz and might get inconsistent answers.", r.URL.Path, r.UserAgent())
```

Comment:
why not as `klog.Warningf` ?

Reply:
copy & paste. But I agree.

Reply:
Actually, no. `V(2).Info` is just right. We still have few clients that make use of this (disaster recovery for example). It's not a warning, just an info.